Detectors/ITSMFT/ITS/tracking/src/VertexerTraits.cxx

```diff
@@ -144,35 +156,74 @@ void VertexerTraits::updateVertexingParameters(const VertexingParameters& vrtPar
   mVrtParams.phiSpan = static_cast<int>(std::ceil(mIndexTableUtils.getNphiBins() * mVrtParams.phiCut /
                                                   constants::math::TwoPi));
   mVrtParams.zSpan = static_cast<int>(std::ceil(mVrtParams.zCut * mIndexTableUtils.getInverseZCoordinate(0)));
+  setNThreads(mVrtParams.nThreads);
 }
 
 void VertexerTraits::computeTracklets()
 {
-  for (int rofId{0}; rofId < mTimeFrame->getNrof(); ++rofId) {
-    trackleterKernelSerial<TrackletMode::Layer0Layer1>(
-      mTimeFrame->getClustersOnLayer(rofId, 0),
-      mTimeFrame->getClustersOnLayer(rofId, 1),
-      mTimeFrame->getIndexTable(rofId, 0).data(),
-      mVrtParams.phiCut,
-      mTimeFrame->getTracklets()[0],
-      mTimeFrame->getNTrackletsCluster(rofId, 0),
-      mIndexTableUtils,
-      rofId,
-      mVrtParams.maxTrackletsPerCluster);
-    trackleterKernelSerial<TrackletMode::Layer1Layer2>(
-      mTimeFrame->getClustersOnLayer(rofId, 2),
-      mTimeFrame->getClustersOnLayer(rofId, 1),
-      mTimeFrame->getIndexTable(rofId, 2).data(),
-      mVrtParams.phiCut,
-      mTimeFrame->getTracklets()[1],
-      mTimeFrame->getNTrackletsCluster(rofId, 1),
-      mIndexTableUtils,
-      rofId,
-      mVrtParams.maxTrackletsPerCluster);
-    mTimeFrame->getNTrackletsROf(rofId, 0) = std::accumulate(mTimeFrame->getNTrackletsCluster(rofId, 0).begin(), mTimeFrame->getNTrackletsCluster(rofId, 0).end(), 0);
-    mTimeFrame->getNTrackletsROf(rofId, 1) = std::accumulate(mTimeFrame->getNTrackletsCluster(rofId, 1).begin(), mTimeFrame->getNTrackletsCluster(rofId, 1).end(), 0);
+
+#pragma omp parallel num_threads(mNThreads)
+  {
+#pragma omp for schedule(dynamic)
```

Comment:
please prefer direct action `#pragma omp parallel for num_threads(mNThreads)`

Reply:
I actually thought that we might keep the threads alive between the two for loops so that we save spawning time (see above comment). I saw a slight perfromance improvement in doing so.
On the other hand, to spawn few threads does no really imply lot of overhead.
If we want to keep it simple it's fine.